online_meetings/canvas_zoom_meetings.py

```diff
@@ -0,0 +1,139 @@
+# Script to get all sites where Zoom is visible and retrieve the meetings to generate a report
+
+import json
+import yaml
+import re
+import os
+import sys
+import requests
+import logging
+import http.client
+from datetime import datetime
+
+from canvasapi import Canvas
+from bs4 import BeautifulSoup as bs
+
+import pandas as pd
+
+# read configurations
+try:
+    with open(os.path.join('../config', 'env.json')) as env_file:
+        ENV = yaml.load(env_file.read())
+except FileNotFoundError:
+    sys.exit(
+        'Configuration file could not be found; please add env.yaml to the config directory.')
+
+LOG_LEVEL = ENV.get('LOG_LEVEL', 'DEBUG')
+logging.basicConfig(level=LOG_LEVEL)
+
+# These two lines enable debugging at httplib level (requests->urllib3->http.client)
+# You will see the REQUEST, including HEADERS and DATA, and RESPONSE with HEADERS but without DATA.
+# The only thing missing will be the response.body which is not logged.
+if LOG_LEVEL == logging.DEBUG:
+    http.client.HTTPConnection.debuglevel = 1
+
+# You must initialize logging, otherwise you'll not see debug output.
+logging.basicConfig(level=LOG_LEVEL)
+logger = logging.getLogger(__name__)
+logger.setLevel(LOG_LEVEL)
+
+requests_log = logging.getLogger("requests.packages.urllib3")
+requests_log.setLevel(LOG_LEVEL)
+requests_log.propagate = True
+# Global variable to access Canvas
+CANVAS = Canvas(ENV.get("CANVAS_URL"), ENV.get("CANVAS_TOKEN"))
+
+
+def zoom_course_report(canvas_account=1, enrollment_term_id=1, published=True):
+    zoom_courses = []
+    zoom_courses_meetings = []
+
+    account = CANVAS.get_account(canvas_account)
+    # Canvas has a limit of 100 per page on this API
+    per_page = 100
+    # Get all published courses from the defined enrollment term
+    courses = account.get_courses(enrollment_term_id=enrollment_term_id, published=published, per_page=per_page)
```

Comment:
This will probably take a few minutes, and I'm essentially doing the same query already in `inventory.py`, with some extra parameters. Thoughts on pulling this from the database at some point? (Fine keeping this as is for now.)

Reply:
Yeah, I was thinking of caching the courses in the database but there's a chance someone might go from published to unpublished right? I think it would be great to eventually replace this with GraphQL but I don't see any way of filtering on published there? This entire process takes about 30 minutes for this term.

Reply:
Jennifer reminded me this morning that in the course inventory report, those courses with less than two students enrollments are filtered out.

We may need to work on the same list of filtered courses, just to be consistent, and ready for justify the results.

Reply:
I don't see in the inventory code where this is happening. Do we have any idea of how many this is? It seems like for zoom meetings you'd want to get the meeting ids and counts no matter the number of students as it could be used for other purposes?

Reply:
This seems to be very cool as it is getting all the courses, without us exclusive about the pagination logic

Reply:
can you time this call? so that in logs we know how long it pulls this info

Reply:
The filtering is happening on the Tableau side right now, but I've integrated the filter into my PR to limit the number of enrollments and users I have to query for via the API. See [this comparison](https://github.com/tl-its-umich-edu/course-inventory/compare/master...ssciolla:issues-23-28-api-plus-sections-minus-wid).

Reply:
The filtering cuts the number of courses more than in half.

Reply:
Since the filtering is done on the Tableau side now, and Sam is working on the embedded the logic into course inventory code, I will close this comment now, and will revisit in the future.